pycheribuild/projects/cross/freertos.py

```diff
@@ -47,17 +47,35 @@ class BuildFreeRTOS(CrossCompileAutotoolsProject):
     needs_sysroot = False  # We don't need a complete sysroot
     supported_architectures = [
         CompilationTargets.BAREMETAL_NEWLIB_RISCV64_PURECAP,
-        CompilationTargets.BAREMETAL_NEWLIB_RISCV64]
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV64,
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV32_PURECAP,
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV32]
     default_install_dir = DefaultInstallDir.SYSROOT
 
     # FreeRTOS Demos to build
     supported_freertos_demos = [
```

Comment:
I do feel each demo should probably be a separate Project, deriving from a common base class, and probably also each app should be a separate Project deriving from the demo's. Having to use options to specify which you want isn't a great interface.

Reply:
That's probably nicer. However, each added projects incurs startup overheads due to all the command line flags that need to be added to argparse. If this is rarely used, I'd prefer making it a single option instead of a separate project.